src/app/integration_test/dad_jokes_page_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:app/access/mocking/mocking_repository.dart';
+import 'package:app/app.dart';
+import 'package:app/app_router.dart';
+import 'package:app/main.dart';
+import 'package:app/presentation/dad_jokes/dad_joke_list_item.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+import 'package:integration_test/integration_test.dart';
+
+Future<void> main() async {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  await initializeComponents(isMocked: true);
+
+  tearDownAll(
+    () async => await GetIt.I.get<MockingRepository>().setMocking(false),
+  );
+
+  testWidgets('Get Dad Jokes', (WidgetTester tester) async {
+    // Arrange
+
+    // Act
+    await tester.pumpWidget(const App());
+
+    // Without this the dadjokesContainer isn't there yet.
+    await tester.pumpAndSettle();
+
+    // Assert
+    var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+    var dadJokes = find.descendant(
+      of: dadjokesContainer,
+      matching: find.byType(DadJokeListItem),
+    );
+
+    expect(dadJokes, findsAtLeast(1));
+  });
+
+  group("Favorites", () {
+    testWidgets('Can remove joke from favorite', (WidgetTester tester) async {
+      // Arrange
+      await tester.pumpWidget(const App());
+      await tester.pumpAndSettle();
+
+      var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+      var dadJokes = find.descendant(
+        of: dadjokesContainer,
+        matching: find.byType(DadJokeListItem),
+      );
+
+      await tester.tap(dadJokes.first);
```

Comment:
I think this test would be better if the favorites were already present (so that you don't add to add to favorites first).

Reply:
I changed it so there's already 1 mocked favorites and it just removes it for the test